stdlib/LinearAlgebra/test/cholesky.jl

```diff
@@ -399,4 +404,25 @@ end
     @test B ≈ Matrix(C)
 end
 
+@testset "CholeskyPivoted and Factorization" begin
+    A = randn(8,8)
+    B = A'A
+    C = cholesky(B, Val(true), check=false)
+    @test CholeskyPivoted{eltype(C)}(C) === C
+    @test Factorization{eltype(C)}(C) === C
+    @test Array(CholeskyPivoted{complex(eltype(C))}(C)) == Array(cholesky(complex(B), Val(true), check=false))
+    @test Factorization{complex(eltype(C))}(C) == cholesky(complex(B), Val(true), check=false)
```

Comment:
These are tough exact equality tests. Is that safe, or vulnerable to floating point issues that we may need to relax later?

Reply:
I only now saw that these tests indeed fail for the obvious reason, so we should really relax them.

Reply:
Oh whoops they passed locally